app/services/code_climate/link.rb

```diff
@@ -0,0 +1,52 @@
+module CodeClimate
+  class Link < BaseService
+    CODE_CLIMATE_API_ORG_NAME = ENV['CODE_CLIMATE_API_ORG_NAME']
+
+    def initialize(project)
+      @project = project
+    end
+
+    def call
+      return if code_climate_project_metric.blank? || !ids_differ
+
+      code_climate_project_metric.update!(cc_repository_id: external_cc_repository_id)
+
+      Rails.logger.info { message }
+    end
+
+    private
+
+    attr_reader :project
+
+    def code_climate_project_metric
+      @code_climate_project_metric ||= CodeClimateProjectMetric.find_by(project_id: project.id)
+    end
+
+    def ids_differ
+      external_cc_repository_id.present? && local_cc_repository_id != external_cc_repository_id
+    end
+
+    def local_cc_repository_id
+      @local_cc_repository_id ||= code_climate_project_metric.cc_repository_id
+    end
+
+    def external_cc_repository_id
+      @external_cc_repository_id ||= respository_by_slug.json['id']
+    end
+
+    def respository_by_slug
+      CodeClimate::Api::Client.new.repository_by_slug(github_slug: github_slug)
+    end
+
+    def github_slug
+      "#{CODE_CLIMATE_API_ORG_NAME}/#{project.name}"
+    end
+
+    def message
+      "PROJECT #{code_climate_project_metric.project_id}"\
```

Comment:
I'd use a HEREDOC for long messages https://www.rubyguides.com/2018/11/ruby-heredoc/